src/main/java/it/pagopa/pm/gateway/service/CcResumeService.java

```diff
@@ -0,0 +1,196 @@
+        String status = CREATED.name();
+        String responseType = StringUtils.EMPTY;
+        String acsUrl = StringUtils.EMPTY;
+        String correlationId = StringUtils.EMPTY;
+        boolean isToAccount = false;
+        switch (resultCode) {
+            case RESULT_CODE_AUTHORIZED:
+                responseType = response.getResponseType().name();
+                isToAccount = true;
+                correlationId = ((ThreeDS2Authorization) response.getThreeDS2ResponseElement()).getTransactionId();
+                break;
+            case RESULT_CODE_CHALLENGE:
+                responseType = response.getResponseType().name();
+                acsUrl = ((ThreeDS2Challenge) response.getThreeDS2ResponseElement()).getAcsUrl();
+                correlationId = ((ThreeDS2Challenge) response.getThreeDS2ResponseElement()).getThreeDSTransId();
+                break;
+            default:
+                log.error(String.format("Error resultCode %s from Vpos for requestId %s", resultCode, entity.getGuid()));
+                status = DENIED.name();
+        }
+        entity.setCorrelationId(correlationId);
+        entity.setStatus(status);
+        entity.setAuthorizationUrl(acsUrl);
+        entity.setResponseType(responseType);
+        paymentRequestRepository.save(entity);
+        return isToAccount;
+    }
+
+    private void executeAccount(PaymentRequestEntity entity, StepZeroRequest pgsRequest) {
+        try {
+            log.info("Calling VPOS - Accounting - for requestId: " + entity.getGuid());
+            Map<String, String> params = vPosRequestUtils.buildAccountingRequestParams(pgsRequest, entity.getCorrelationId());
+            HttpClientResponse clientResponse = callVPos(params);
+            AuthResponse response = vPosResponseUtils.buildAuthResponse(clientResponse.getEntity());
+            vPosResponseUtils.validateResponseMac(response.getTimestamp(), response.getResultCode(), response.getResultMac(), pgsRequest);
+            checkAccountResultCode(response, entity);
+        } catch (Exception e) {
+            log.error(GENERIC_ERROR_MSG + entity.getIdTransaction() + " stackTrace: " + Arrays.toString(e.getStackTrace()));
+        }
+        executePatchTransaction(entity);
+    }
+
+    private void checkAccountResultCode(AuthResponse response, PaymentRequestEntity entity) {
+        String resultCode = response.getResultCode();
+        String status = AUTHORIZED.name();
+        boolean authorizationOutcome = true;
+        if (!resultCode.equals(RESULT_CODE_AUTHORIZED)) {
+            status = DENIED.name();
+            authorizationOutcome = false;
+        }
+        entity.setAuthorizationOutcome(authorizationOutcome);
+        entity.setStatus(status);
+        paymentRequestRepository.save(entity);
+        log.info("END - XPay Request Payment Account for requestId " + entity.getGuid());
+    }
+
+    private void executePatchTransaction(PaymentRequestEntity entity) {
```

Comment:
Maybe this method should be updated with the logic to call eCommerce API
`PATCH /transactions/{transactionId}/auth-requests`
currently on development with task https://pagopa.atlassian.net/browse/PM-1257

@giovanniberti do you agree?

Reply:
That's right, the method shoulg be updated.
I think we can create a task on purpose.